### DISCUSSION TITLE
documentation issue: requires custom libtorrent

### DISCUSSION THREAD

**Discussion by lmfont** [jesec/rtorrent#9] [General]
As per my comment [here](https://github.com/jesec/rtorrent/discussions/7#discussioncomment-365959), I was going to try to write a PKGBUILD for this rtorrent fork. I wanted to use cmake because a dynamic build that uses libraries managed by the package manager would be ideal for an Arch package.

The readme says:

> libtorrent with development files (core dependency, matching version required)

But this suggests that I should be able to just use the correct system vendored libtorrent (0.13.8), it's not clear that I need to use a fork (or that one even exists). Took me some time to figure this out.

For example, `/usr/include/torrent/utils/ranges.h` tries to include `rak/functional.h`, but you've removed this from your rtorrent fork, and your version of libtorrent no longer includes it.

Of course this issue doesn't show up for the bazel build, but I don't have as much interest in a PKGBUILD that pulls in a bunch of internet dependencies for a static build. I think what this means is that a PKGBUILD would need to depend on something like libtorrent-jesec, which would be a separate PKGBUILD in the AUR.

(Sorry if I somehow misunderstood this issue, I mainly work with high level languages like Python.)

**Comment by jesec** [comment]
Since this is a question, I am moving it to discussions. 

In my opinion, the documentation is quite clear. It is simply natural to use a matching revision of libTorrent. 

Nonetheless, I will add a link to my libTorrent repo.

**Reply by lmfont** [reply]
The readme says "matching version required", not "matching revision". I'm not sure how anyone who didn't know you forked libtorrent would possibly guess that. "Matching version" and "matching revision" are both ambiguous, regardless. It's generally understood that releases of (vanilla) rtorrent rely on specific versions of (vanilla) libtorrent, and so I assumed that's what this message was about. In other words, that you wanted to avoid users trying to compile your rtorrent fork on Debian stable, which only includes libtorrent 0.13.7, not 0.13.8.

Maybe this is just a difference in mentality, since users who are used to bazel and the static linking approach are going to expect whatever specific libtorrent is needed to be pulled in automatically, which might or might not be the same as the one provided by their system.
